### PR TITLE
MBS-12598 Add emcee QueueApi.kt

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

File: subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Device.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import com.squareup.moshi.Json
+
+public data class Device(
+    /**
+     * Should be deleted
+     */
+    @Json(name = "deviceType")
+    val type: String,
+    /**
+     * Representing a emulator api version i.e. 21,22,29,30 etc.
```
Comment (eugene-krivobokov) [inline subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Device.kt]:
```suggestion
     * Representing an emulator API version: 21,22,29,30 etc.
```

File: subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Device.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+     */
```
Comment (eugene-krivobokov) [inline subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Device.kt:13]: Let's use `@Json` for all json fields.
It's too easy to break a contract while refactoring.

File: subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Job.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.avito.emcee.queue
+
+import com.squareup.moshi.Json
+
+public data class Job(
+    @Json(name = "jobId")
+    val id: String,
+    /**
+     * Uses to prioritize inside the group. Higher will schedule faster
```
Comment (eugene-krivobokov) [inline subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Job.kt:9]: Are all integer values valid?

File: subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/TestTimeoutConfiguration.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.avito.emcee.queue
+
+public data class TestTimeoutConfiguration(
+    val singleTestMaximumDuration: Int,
+    val testRunnerMaximumSilenceDuration: Int
```
Comment (eugene-krivobokov) [inline subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/TestTimeoutConfiguration.kt:5]:
```suggestion
    val testRunnerMaximumSilenceDurationSec/Ms/...: Int
```
Let's clarify units if we can't use better types here and convert them by moshi.

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/TestTimeoutConfiguration.kt (hunk 4 above)
Comment (sergioboi) [reply subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/TestTimeoutConfiguration.kt:5]: This part will be changed

File: subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Job.kt (hunk 3 above)
Comment (sergioboi) [reply subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Job.kt:9]: @beefon this question is for you

File: subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Device.kt (hunk 2 above)
Comment (sergioboi) [reply subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Device.kt:13]: I think it's an overhead. I think we will rarely change those models. And they are placed in a separate module

Comment (eugene-krivobokov) [reply subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Device.kt:13]: The motivation is don't even think about it while changing any code

**Review by dsvoronin — APPROVED** [review]

File: subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Job.kt (hunk 3 above)
Comment (beefon) [reply subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/Job.kt:9]: Expected to be in range of `[0, 999]`

File: subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/TestExecutionBehavior.kt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.avito.emcee.queue
+
+import com.squareup.moshi.Json
+
+public data class TestExecutionBehavior(
+    val environment: Map<String, String>,
+    @Json(name = "numberOfRetries")
+    val retries: Int
```
Comment (beefon) [inline subprojects/emcee/queue-api/src/main/kotlin/com/avito/emcee/queue/TestExecutionBehavior.kt:8]: There are some new fields out there
https://github.com/avito-tech/Emcee/blob/master/Sources/RunnerModels/TestExecutionBehavior.swift

If you don't need them, I guess you just don't parse them.

**Pull request merged**